supervisord.conf

```diff
@@ -21,7 +24,13 @@ stdout_logfile = /var/log/digitalmarketplace/application.log
 stdout_logfile_maxbytes = 50000000
 stdout_logfile_backups = 3
 redirect_stderr = true
-stopsignal = INT
+# USR1 doesn't actually stop uwsgi at all - it prints statistics (!), but we want to hang on as long
+# as possible after bosh asks us to shut down to give us a chance to handle as much of the potential
+# connection backlog we might have.
+stopsignal = USR1
+# bosh will probably kill everything after 10s - stop a second early in the hopes that the logs of our
+# last handled request make it off the machine
+stopwaitsecs = 9
```

Comment:
Why is this better than `gracefully_kill_them_all`?

Reply:
In circumstances where we have a big backlog `gracefully_kill_them_all` will do nothing for us, because it will wait for the whole backlog to be drained, which (again, in extreme circumstances) will take more than the 10s bosh is willing to give us.